junit/src/main/java/org/kaazing/k3po/junit/rules/ScriptRunner.java

```diff
@@ -214,6 +214,7 @@ public void notifyBarrier(final String barrierName) throws InterruptedException
 
 
     public void dispose() throws Exception {
+        latch.awaitFinished();
```

Comment:
First of all, even without this change, we are still waiting for `latch.awaitFinished()` in K3poRule. So if it will get stuck, the test will get stuck anyway. It does not matter if I'm adding one more call to `latch.awaitFinished()`.

But one of the goals of the changes I did all this year is to have that communication thread stuck only in extremely rare situations.  If you look at `ScriptRunner.call()`, there is only one exception being caught inside the loop - `SocketTimeoutException`. Any other exception will end the loop and the thread will end, with a finally that will notify the finished latch. 

The change I did in this PR is for the following reason: I have found that in a case of an exception in `base.evaluate()`, first we call `scriptRunner.abort()` and then in the finally we call `scriptRunner.dispose()`. Now that we eliminated the DISPOSE command, method `dispose()` is just closing the connection. Sometimes the closing of the connection was happening before the communication thread managed to send the ABORT and wait for the FINISHED to come. It was just like not calling abort() at all.